target/linux/ipq40xx/files-5.4/arch/arm/boot/dts/qcom-ipq4019-wtr-m2133hp.dts

```diff
@@ -0,0 +1,413 @@
+				"gpio59";
+			function = "qpic";
+			bias-pull-up;
+		};
+
+		pulldowns {
+			pins = "gpio54", "gpio55", "gpio56",
+				"gpio57", "gpio60", "gpio61",
+				"gpio62", "gpio63", "gpio64",
+				"gpio65", "gpio66", "gpio67",
+				"gpio68", "gpio69";
+			function = "qpic";
+			bias-pull-down;
+		};
+	};
+};
+
+&blsp_dma {
+	status = "ok";
+};
+
+&blsp1_uart1 {
+	pinctrl-0 = <&serial_0_pins>;
+	pinctrl-names = "default";
+	status = "ok";
+};
+
+&cryptobam {
+	status = "ok";
+};
+
+&pcie0 {
+	status = "ok";
+
+	bridge@0,0 {
+		reg = <0 0 0 0 0>;
+
+		wifi@0,0 {
+			compatible = "qcom,ath10k";
+			reg = <0 0 0 0 0>;
+
+			qcom,ath10k-calibration-variant = "Buffalo-WTR-M2133HP";
+		};
+	};
+};
+
+&qpic_bam {
+	status = "ok";
+};
+
+&nand {
+	pinctrl-0 = <&nand_pins>;
+	pinctrl-names = "default";
+	status = "ok";
+
+	nand@0 {
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "SBL1";
+				reg = <0x0000000 0x0100000>;
+				read-only;
+			};
+
+			partition@100000 {
+				label = "MIBIB";
+				reg = <0x0100000 0x0100000>;
+				read-only;
+			};
+
+			partition@200000 {
+				label = "BOOTCONFIG";
+				reg = <0x0200000 0x0100000>;
+				read-only;
+			};
+
+			partition@300000 {
+				label = "QSEE";
+				reg = <0x0300000 0x0100000>;
+				read-only;
+			};
+
+			partition@400000 {
+				label = "QSEE_1";
+				reg = <0x0400000 0x0100000>;
+				read-only;
+			};
+
+			partition@500000 {
+				label = "CDT";
+				reg = <0x0500000 0x0080000>;
+				read-only;
+			};
+
+			partition@580000 {
+				label = "CDT_1";
+				reg = <0x0580000 0x0080000>;
+				read-only;
+			};
+
+			partition@600000 {
+				label = "BOOTCONFIG1";
+				reg = <0x0600000 0x0080000>;
+				read-only;
+			};
+
+			partition@680000 {
+				label = "APPSBLENV";
+				reg = <0x0680000 0x0080000>;
+			};
+
+			partition@700000 {
+				label = "APPSBL";
+				reg = <0x0700000 0x0200000>;
+				read-only;
+			};
+
+			partition@900000 {
+				label = "APPSBL_1";
+				reg = <0x0900000 0x0200000>;
+				read-only;
+			};
+
+			art: partition@b00000 {
+				label = "ART";
+				reg = <0x0b00000 0x0080000>;
+				read-only;
+			};
+
+			partition@b80000 {
+				label = "ART_1";
+				reg = <0x0b80000 0x0080000>;
+				read-only;
+			};
+
+			orgdata: partition@c00000 {
+				label = "ORGDATA";
+				reg = <0x0c00000 0x0080000>;
+				read-only;
+			};
+
+			partition@c80000 {
+				label = "ORGDATA_1";
+				reg = <0x0c80000 0x0080000>;
+				read-only;
+			};
+
+			partition@d00000 {
+				label = "ubi";
+				reg = <0x0d00000 0x2900000>;
+			};
+
+			partition@3600000 {
+				label = "rootfs_recover";
+				reg = <0x3600000 0x2900000>;
+				read-only;
+			};
+
+			partition@5f00000 {
+				label = "user_property";
+				reg = <0x5f00000 0x1a20000>;
+				read-only;
+			};
+		};
+	};
+};
+
+&wifi0 {
+	status = "ok";
+	qcom,ath10k-calibration-variant = "Buffalo-WTR-M2133HP";
+	ieee80211-freq-limit = <2400000 2483000>;
```

Comment:
All other devices in OpenWrt use 2500000, I don't see any particular reason to use otherwise here. And it seems no ipq40xx device uses `ieee80211-freq-limit` for 2.4GHz band, do we actually need this?

Reply:
Without this, 2.4GHz 14ch is enabled. However the hardware does not support the use of 14ch, so we need to disable it manually.

Reply:
Okay, thanks for your clarification.

Reply:
I'm not sure if it's necessary, but if two 5GHz chips support different channels, then we might be able to apply `ieee80211-freq-limit` there as well, what do you think?
```
$ git grep "ieee80211-freq-limit" target/linux/ipq40xx/files-5.4
target/linux/ipq40xx/files-5.4/arch/arm/boot/dts/qcom-ipq4019-a62.dts:                  ieee80211-freq-limit = <5170000 5350000>;
target/linux/ipq40xx/files-5.4/arch/arm/boot/dts/qcom-ipq4019-a62.dts:  ieee80211-freq-limit = <5470000 5875000>;
target/linux/ipq40xx/files-5.4/arch/arm/boot/dts/qcom-ipq4019-ea8300.dts:       ieee80211-freq-limit = <5170000 5330000>;
target/linux/ipq40xx/files-5.4/arch/arm/boot/dts/qcom-ipq4019-ea8300.dts:       ieee80211-freq-limit = <5490000 5835000>;
target/linux/ipq40xx/files-5.4/arch/arm/boot/dts/qcom-ipq4019-eap2200.dts:                      ieee80211-freq-limit = <5470000 5875000>;
target/linux/ipq40xx/files-5.4/arch/arm/boot/dts/qcom-ipq4019-eap2200.dts:      ieee80211-freq-limit = <5170000 5350000>;
target/linux/ipq40xx/files-5.4/arch/arm/boot/dts/qcom-ipq4019-fritzrepeater-3000.dts:   ieee80211-freq-limit = <5170000 5350000>;
target/linux/ipq40xx/files-5.4/arch/arm/boot/dts/qcom-ipq4019-fritzrepeater-3000.dts:                   ieee80211-freq-limit = <5470000 5875000>;
target/linux/ipq40xx/files-5.4/arch/arm/boot/dts/qcom-ipq4019-map-ac2200.dts:   ieee80211-freq-limit = <5470000 5875000>;
target/linux/ipq40xx/files-5.4/arch/arm/boot/dts/qcom-ipq4019-map-ac2200.dts:                   ieee80211-freq-limit = <5170000 5350000>;
```

Reply:
Two 5GHz chips are correctly limited their frequency without "freq-limit". It seems this limitation comes from custom BDFs.

Reply:
Okay, that's nice.